crates/bifrost/src/watchdog.rs

```diff
@@ -51,6 +60,8 @@ impl Watchdog {
             sender,
             inbound,
             live_providers: Vec::with_capacity(ProviderKind::LENGTH),
+            in_flight_trim: None,
+            pending_trims: HashMap::with_capacity(128),
```

Comment:
Nitpicking on the final pass, but I'm wondering if there's a sensible config parameter we can use instead of a magic number. Number of logs or partitions from either config or metadata?